src/auth_and_perms/api/serializers.py

```diff
@@ -174,4 +176,39 @@ def validate(self, data):
             logger.debug(
                 f'ValidateUserAccessOrgLabSerializer --> check_user_access_kwargs_org_lab is ({check_user_access})')
             raise serializers.ValidationError({'user': _("User doesn't have permissions")})
+
+        if shelf_object:
+            if shelf_object.in_where_laboratory != laboratory:
+                logger.debug(
+                    f'ValidateUserAccessOrgLabSerializer --> shelfobject.in_where_laboratory!=laboratory')
+                raise serializers.ValidationError(
+                    {'shelfobject': _("Shelfobject does not belongs to this laboratory.")})
+
+            if shelf_object.in_where_laboratory.organization != organization:
+                logger.debug(
+                    f'ValidateUserAccessOrgLabSerializer --> shelfobject.in_where_laboratory.organization!= organization')
+                raise serializers.ValidationError(
+                    {'shelfobject': _("Shelfobject does not belongs to this organization.")})
+
+            if shelf:
+                if shelf != shelf_object.shelf:
+                    logger.debug(
+                        f'ValidateUserAccessOrgLabSerializer --> shelf != shelf_object.shelf')
+                    raise serializers.ValidationError(
+                        {'shelfobject': _(
+                            "Shelfobject does not belongs to that shelf.")})
```

Comment:
It should probably use "this shelf" instead of "that shelf" here, to follow conventions... please review anywhere else this message might be used to change those places as well so we don't have both "this" and "that" in the translations file.